src/audio-manager.ts

```diff
@@ -0,0 +1,226 @@
+import {_audioContext} from './audio-listener.js';
+
+const preloadedBuffers: {[key: string]: [Promise<AudioBuffer>, number]} = {};
+
+/** AudioManager loads and manages audiofiles from which PlayableNodes are created
+ * @example
+ * ```
+ * start() {
+ *      this.audio = await AudioManager.load('click.wav');
+ * }
+ * onPress() {
+ *      this.audio.play();
+ * }
+ * // if not needed any longer
+ * this.audio.destroy();
+ * ```
+ */
+export const AudioManager = {
+    /**
+     * Asynchronously loads an audio file and returns a Promise that resolves with a PlayableNode.
+     *
+     * @note Make sure to load files on `start()`, so that Nodes are ready when they are needed.
+     *
+     * @param file - The path or URL of the audio file to be loaded.
+     * @returns A Promise that resolves with a PlayableNode representing the loaded audio.
+     * @throws If there is an error during the loading process, a rejection with an error message is returned.
+     */
+    async load(file: string): Promise<PlayableNode> {
+        try {
+            /* Return when a instance of this file already is being loaded */
+            const [bufferPromise, referenceCount] = preloadedBuffers[file] || [
+                undefined,
+                0,
+            ];
+            if (bufferPromise !== undefined) {
+                preloadedBuffers[file][1] += 1;
+                return bufferPromise.then(() => new PlayableNode(file));
+            }
+            const response = await fetch(file);
+
+            if (!response.ok) {
+                return Promise.reject(`Failed to fetch audio data from ${file}`);
+            }
+
+            /* Create promise that resolves once decoding is complete */
+            const decodingPromise = new Promise<AudioBuffer>((resolve, reject) => {
```

Comment:
I would implement the entire logic a little differently. Store a list of temporary callbacks that resolve a promise for each file, those are basically listening to a file to be loaded. Once a file is loaded, go trough each listener and resolve the promise. Then each promise will be resolved witht the exact same result, isn't that what was orignally expected?